src/charm.py

```diff
@@ -179,7 +190,8 @@ def _on_config_changed(self, event):
         retry=retry_if_exception_type(ApiError),
         reraise=True,
         before=before_log(logger, logging.DEBUG),
-        wait=wait_exponential(multiplier=1, min=2, max=60 * 2),
+        wait=wait_exponential(multiplier=1, min=2, max=15),
+        stop=stop_after_delay(60 * 5),
```

Comment:
So long as the `ApiError` exception occurs, wait seconds (2, 4, 8, 15 .. 15 ... 15 ) for a maximum of `5m`